src/test/ts/part2/Exercise2ArrayFunctionsTest.ts

```diff
@@ -8,26 +8,35 @@ describe('Exercise2ArrayFunctionsTest', () => {
   });
 
   it('frog names', () => {
-    assert.deepEqual(Ex.frogNames(Ex.myFrogs), [ 'frog1', 'frog2', 'loudfrog', 'quietfrog' ], 'frog names');
+    assert.deepEqual(Ex.frogNames(Ex.myFrogs), ['frog1', 'frog2', 'loudfrog', 'quietfrog'], 'frog names');
   });
 
   it('frog ages', () => {
     // TODO: write a test for your frog ages function
+    assert.deepEqual(Ex.frogAges(Ex.myFrogs), [3, 4, 1, 10], 'frog names');
   });
 
   it('ribbitting frogs', () => {
     assert.deepEqual(Ex.ribbitting(Ex.myFrogs), [
-      { name: 'frog1', ribbits: true, age: 3 },
-      { name: 'loudfrog', ribbits: true, age: 1 }
+      {name: 'frog1', ribbits: true, age: 3},
+      {name: 'loudfrog', ribbits: true, age: 1}
     ], 'ribbitting');
   });
 
   it('older frogs', () => {
     assert.deepEqual(Ex.olderFrogs(Ex.myFrogs), [
-      { name: 'quietfrog', ribbits: false, age: 10 }
+      {name: 'quietfrog', ribbits: false, age: 10}
     ], 'older frogs');
   });
 
+  it('ribbitting frog exists', () => {
+    assert.deepEqual(Ex.ribbittingFrogExists(Ex.myFrogs), true, 'ribbittingFrogExists');
+  });
+
+  it('negative exists', () => {
+    assert.deepEqual(Ex.negativeExists([1, 2, -1]), true, 'negativeExists');
```

Comment:
Same here this could be assert.isTrue or at least assert.equal since a boolean doesn't have a nested structure deepEqual is a bit overkill though I doubt it would be that much of a performance difference but. It's better to describe the intent.